File: Frontend/exam-portal-fe/src/app/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {

  constructor(private _httpClient: HttpClient) { }

  public getAllCategories() {
    return this._httpClient.get(`${baseUrl}/api/categories`);
  }

  public addCategory(category: any) {
    return this._httpClient.post(`${baseUrl}/api/categories`, category);
  }

  public deleteCategory(cid: any) {
    return this._httpClient.delete(`${baseUrl}/api/categories/${cid}`);
  }


  public getCategoryById(cid: any) { 
    return this._httpClient.get(`${baseUrl}/api/categories/${cid}`);
  }

  public updateCategory(category: any) {
    return this._httpClient.put(`${baseUrl}/api/categories`, category);
  }
}

File: Backend/exam-server/src/main/java/com/exam/examserver/service/CategoryService.java
package com.exam.examserver.service;

import com.exam.examserver.entity.Category;

import java.util.List;

public interface CategoryService {

    Category addCategory(Category category);

    Category updateCategory(Category category);

    List<Category> getAllCategories();

    Category getByCategoryId(long cId);

    void deleteByCategoryId(long cId);
}
